src/transformStateWithClones.js

```diff
@@ -8,6 +8,38 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const clone = [];
+  const copyState = { ...state };
+  let copy = {};
+
+  actions.forEach(action => {
+    switch (true) {
+      case action.type === 'addProperties':
+        copy = Object.assign(copyState, action.extraData);
+        clone.push({ ...copy });
+        break;
+      case action.type === 'removeProperties':
+        action.keysToRemove.forEach(remove => {
+          for (const value in copyState) {
+            if (value === remove) {
+              delete copyState[value];
+            }
+          }
+        });
+        copy = copyState;
+        clone.push({ ...copy });
```

Comment:
You can push just once on each iteration

Reply:
Fix it 
![image](https://user-images.githubusercontent.com/97547824/202766808-0b499542-643a-4d46-b110-cfcaa10b4b97.png)